### PR TITLE
use configurable functionsDir and publishDir throughout NoN

### PR DESCRIPTION
**Pull request by lindsaylevine**
cc: @FinnWoelm 

this is needed for https://github.com/netlify/netlify-plugin-nextjs/issues/43

### CODE REVIEW

File: lib/helpers/getNextConfig.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const { default: loadConfig } = require("next/dist/next-server/server/config");
+const { resolve } = require("path");
+
+const getNextConfig = () => {
+  // Load next.config.js
+  // Use same code as https://github.com/vercel/next.js/blob/25488f4a03db30cade4d086ba49cd9a50a2ac02e/packages/next/build/index.ts#L114
+  return loadConfig(PHASE_PRODUCTION_BUILD, resolve("."));
+};
+
+module.exports = getNextConfig;
```
Comment (lindsaylevine) [inline lib/helpers/getNextConfig.js:12]: i thought i'd want and need this file in another part but ended up not using it here (except calling from getNextDistDir). this is a function i'm also adding in the i18n work so we can leave it :)

File: index.js [hunk 2]
```diff
@@ -3,17 +3,30 @@ const copyPublicFiles = require("./lib/steps/copyPublicFiles");
...
-  prepareFolders();
+/** options param:
+ * {
+ *   functionsDir: string to path
+ *   publishDir: string to path
+ * }
+ */
+const nextOnNetlify = (options = {}) => {
+  const functionsPath = options.functionsDir || NETLIFY_FUNCTIONS_PATH;
+  const publishPath = options.pubishDir || NETLIFY_PUBLISH_PATH;
```
Comment (ehmicky) [inline index.js:19]:
```suggestion
  const publishPath = options.publishDir || NETLIFY_PUBLISH_PATH;
```

**Review by ehmicky — APPROVED** [review]
Looks good (except for a small typo), that's amazing!
We might want to update the `README.md` and udpate the two issues asking for this feature too :)

File: index.js (hunk 2 above)
Comment (lindsaylevine) [reply index.js:19]: omg big catch ty

**Comment by lindsaylevine** [comment]
ok yes ill push up a README commit. when updating the issues that have asked for this feature, are you suggesting we point them to the plugin? otherwise their issue isn't really resolved directly through NoN right?

**Comment by ehmicky** [comment]
Maybe we should update the CLI entrypoint to include those two new options as CLI flags? Like this, we could document it in the `README.md` (since that only covers the CLI entrypoint, not programmatic). What do you think?

**Comment by lindsaylevine** [comment]
wont that possibly conflict with UI settings? when i was starting this work i started going down that path but stopped knowing id run into toml vs ui issues. i feel like its easier/safer to just defer to the plugin :/

Screen Shot 2020-11-17 at 1 13 42 PM

**Comment by ehmicky** [comment]
Yes, this makes sense! :+1:

**Pull request merged**